src/frontend/src/lib/components/docs/DocForm.svelte

```diff
@@ -0,0 +1,272 @@
+        <div class="form-doc-id">
+          <input
+            id="doc-id"
+            type="text"
+            placeholder="Document ID"
+            name="doc_id"
+            bind:value={docId}
+          />
+          <button
+            class="primary"
+            on:click|preventDefault={() => docId = generateUniqueDocID()}
+            >
+              {$i18n.document.field_doc_id_btn_auto_id}
+          </button>
+        </div>
+      </div>
+      {#each fields as field, i (i)}
+        <div class="form-field">
+          <div class="form-field-item">
+            <label>{$i18n.document.field_name_label}</label>
+            <input
+              id="field_name"
+              type="text"
+              placeholder={$i18n.document.field_name_placeholder}
+              name="field_name"
+              bind:value={field.name}
+            />
+          </div>
+          <div class="form-field-item">
+            <label>{$i18n.document.field_type_label}</label>
+            <select id="field_type" name="field_type" bind:value={field.fieldType}>
+              <option value={DocFieldTypeEnum.boolean}>{$i18n.document.field_type_boolean}</option>
+              <option value={DocFieldTypeEnum.string}>{$i18n.document.field_type_string}</option>
+              <option value={DocFieldTypeEnum.number}>{$i18n.document.field_type_number}</option>
+            </select>
+          </div>
+          <div class="form-field-item">
+            <div>
+              <label>{$i18n.document.field_value_label}</label>
+              <div class="value-input-wrapper">
+                {#if field.fieldType === DocFieldTypeEnum.number || field.fieldType === DocFieldTypeEnum.string}
+                  <input
+                    id="value"
+                    type={field.fieldType === DocFieldTypeEnum.string ? "text" : "number"}
+                    placeholder={$i18n.document.field_value_label}
+                    name="field_value"
+                  />
+                {/if}
+
+                {#if field.fieldType === DocFieldTypeEnum.boolean} 
+                  <select id="field_value" name="field_value" placeholder="Field value" bind:value={field.value}>
+                    <option value={true}>{$i18n.document.field_value_true}</option>
+                    <option value={false}>{$i18n.document.field_value_false}</option>
+                  </select>
+                {/if}
+
+                {#if fields.length > 1}
+                  <button
+                    class="text action start"
+                    on:click|preventDefault={() => onDeleteFieldPressed(i)}
+                    >
+                      <IconDelete size="24px" />
+                  </button>
+                {/if}
+              </div>  
+            </div>            
+          </div>
+        </div>
+      {/each}
+
+      <button
+        class="text action start"
+        on:click|preventDefault={onAddFieldButtonPressed}
```

Comment:
you can replace the `preventDefault` by setting the `type=button` which is semantically a bit more accurate